### PR TITLE
[#343] improvement(build): Shade Netty Native lib

### PR DESCRIPTION
**Pull request by cchung100m**
Hi @jerqi 

Any comments and suggestions would be appreciated if you are available, thank you.

### What changes were proposed in this pull request?

Add the shade Netty Native lib into` pom.xml`

### Why are the changes needed?

Fix: #343 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

current UT

### CODE REVIEW

**Comment by jerqi** [comment]
We should only shade jars in the framework client like client-spark3, client-spark2, client-mr. You can modify https://github.com/apache/incubator-uniffle/blob/d09cdc22eb77306530c689ef273b51953e9095c2/client-spark/spark3/pom.xml#L96

**Comment by cchung100m** [comment]
> We should only shade jars in the framework client like client-spark3, client-spark2, client-mr. You can modify
> 
> https://github.com/apache/incubator-uniffle/blob/d09cdc22eb77306530c689ef273b51953e9095c2/client-spark/spark3/pom.xml#L96


Hi @jerqi 

Thanks for the prompt reply. I'd like to confirm whether the proposed change is feasible or not.

**Comment by jerqi** [comment]
> > We should only shade jars in the framework client like client-spark3, client-spark2, client-mr. You can modify
> > https://github.com/apache/incubator-uniffle/blob/d09cdc22eb77306530c689ef273b51953e9095c2/client-spark/spark3/pom.xml#L96
> 
> Hi @jerqi
> 
> Thanks for the prompt reply. I'd like to confirm whether the proposed change is feasible or not.

Yes, I think it is feasible.

**Comment by cchung100m** [comment]
Hi @jerqi 

It seems like there are some integration testing errors that occurred, any suggestions would be appreciated if you are available to help.

File: server/pom.xml [hunk 1]
```diff
@@ -203,6 +203,13 @@
                   <pattern>com.fasterxml.jackson.databind</pattern>
                   <shadedPattern>${rss.shade.packageName}.jackson.databind</shadedPattern>
                 </relocation>
+                <relocation>
```
Comment (jerqi) [inline server/pom.xml:206]: Could you revert this change?

Comment (cchung100m) [reply server/pom.xml]: Hi @jerqi Thanks for the kind reminder and suggestions. I updated the part you mentioned.

**Review by jerqi — APPROVED** [review]
LGTM, thanks @cchung100m , merged to master.
